framework/src/play-integration-test/src/test/scala/play/it/api/SecretConfigurationParserSpec.scala

```diff
@@ -0,0 +1,94 @@
+        .configure(extraConfig: _*)
+        .build()
+      val (secret, events) = LogTester.recordLogEvents {
+        app.httpConfiguration.secret.secret
+      }
+      (secret, events)
+    } match {
+      case scala.util.Success((secret, events)) => (Option(secret), events)
+      case scala.util.Failure(_) => (None, Seq.empty)
+    }
+  }
+
+  "When parsing SecretConfiguration" should {
+    "in DEV mode" should {
+      "return 'changeme' when it is configured to it" in {
+        val (secret, events) = parseSecret(mode = Mode.Dev)("play.http.secret.key" -> "changeme")
+        events.map(_.getFormattedMessage).find(_.contains("Generated dev mode secret")) must beSome
+        secret must beSome
+      }
+
+      "generate a secret when no value is configured" in {
+        val (secret, events) = parseSecret(mode = Mode.Dev)("play.http.secret.key" -> null)
+        events.map(_.getFormattedMessage).find(_.contains("Generated dev mode secret")) must beSome
+        secret must beSome
+      }
+
+      "log a warning when secret length is smaller than 8 chars" in {
+        val (secret, events) = parseSecret(mode = Mode.Dev)("play.http.secret.key" -> "1234567")
+        events.map(_.getFormattedMessage).find(_.contains("The application secret is too short and does not have the recommended amount of entropy")) must beSome
+        secret must beSome
+      }
+
+      "log a warning when secret length is smaller then 15 chars" in {
+        val (secret, events) = parseSecret(mode = Mode.Dev)("play.http.secret.key" -> "1234567890123")
+        events.map(_.getFormattedMessage).find(_.contains("Your secret key is very short, and may be vulnerable to dictionary attacks.  Your application may not be secure")) must beSome
+        secret must beSome
+      }
+
+      "return the value without warnings when it is configured respecting the requirements" in {
+        val (secret, events) = parseSecret(mode = Mode.Dev)("play.http.secret.key" -> "12345678901234567890")
+        events.map(_.getFormattedMessage).find(_.contains("Your secret key is very short, and may be vulnerable to dictionary attacks.  Your application may not be secure")) must beNone
+        secret must beSome
+      }
+    }
+
+    "in PROD mode" should {
+      "fail when value is configured to 'changeme'" in {
+        val (secret, _) = parseSecret(mode = Mode.Prod)("play.http.secret.key" -> "changeme")
+        secret must beNone
+      }
+
+      "fail when value is not configured" in {
+        val (secret, _) = parseSecret(mode = Mode.Prod)("play.http.secret.key" -> null)
+        secret must beNone
+      }
+
+      "fail when value length is smaller than 8 chars" in {
+        val (secret, _) = parseSecret(mode = Mode.Prod)("play.http.secret.key" -> "1234567")
+        secret must beNone
+      }
+
+      "log a warning when value length is smaller than 15 chars" in {
+        val (secret, events) = parseSecret(mode = Mode.Prod)("play.http.secret.key" -> "1234567890123")
```

Comment:
```suggestion
        val (secret, events) = parseSecret(mode = Mode.Prod)("play.http.secret.key" -> "123456789012345")
```

I noticed on the code below the actual conditions are not `smaller than` `8` or `15` but `smaller than or equal` so the wording of the test case and the value of the secret need reviewing.

Reply:
Fixed